src/compiler/checker.ts

```diff
@@ -6127,7 +6127,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             },
             isOptionalParameter,
             isUndefinedIdentifierExpression(node: Identifier) {
-                Debug.assert(isExpressionNode(node));

```

Comment:
The identifier being passed in was the `foo` of an `export default foo`, which is an expression, but doesn’t return true in this syntactic check because `export default SomeType` is also legal. I’m not sure if this assertion really needs to be here in any form.

Reply:
Ref #60692